docs: add release article for nuxt v4.5 (#2323)
nuxt/nuxt.com: content/blog/46.v4-5.md

    to: https://bsky.app/profile/danielroe.dev
date: 2026-07-18T10:00:00.000Z
category: Release
---

Nuxt 4.5 is a big one. This release ships three major upgrades to the build layer (Vite 8, Rspack 2, and a brand new Rsbuild-powered pipeline for the Rspack builder), an experimental SSR streaming mode, a handful of new composables and conventions, and a lot of groundwork that brings us closer to Nuxt 5.

There's a lot here, so grab a coffee. ☕️

## 📣 Some News

### Preparing for Nuxt 5

A good chunk of this release is (hopefully) invisible plumbing for Nuxt 5. We've moved onto the latest major versions of several core dependencies ([`unhead` v3](#unhead-v3), [`unctx` v3](#unctx-v3), and [Vite 8](#vite-8)), switched the framework's own build over to [`tsdown`](https://github.com/nuxt/nuxt/pull/35179), and introduced a stable `nuxt/*` build output contract with dev exports so that type-checking in the Nuxt monorepo works without a build step ([#35463](https://github.com/nuxt/nuxt/pull/35463), [#35605](https://github.com/nuxt/nuxt/pull/35605)).

Much of this is working to shrink the gap between v4 and v5 internally, so that the migration will be as boring(?) as possible.

::tip

> If you want to test some of the breaking changes of Nuxt v4,, you can already opt in with `future.compatibilityVersion: 5`. Keep an eye on the [Upgrade Guide](/docs/getting-started/upgrade) for details as they land.

Author: ```suggestion
If you want to test some of the breaking changes of Nuxt v4, you can already opt in with `future.compatibilityVersion: 5`. Keep an eye on the [Upgrade Guide](/docs/getting-started/upgrade) for details as they land.
```

:read-more{to="/docs/api/composables/use-layout"}

## 🪟 Named Views

Nuxt now supports **named views** through a filename convention ([#35123](https://github.com/nuxt/nuxt/pull/35123)). If a parent page renders more than one `<NuxtPage>` outlet, you can give each outlet a name and provide a sibling page file for it using the `name@view.vue` convention:

```bash [Directory Structure]
-| pages/
---| parent/
-----| child.vue
-----| child@sidebar.vue
---| parent.vue
```

```vue [pages/parent.vue]
<template>
  <div>
    <NuxtPage />
    <aside>
      <NuxtPage name="sidebar" />
    </aside>
  </div>
</template>
```

Navigating to `/parent/child` renders `child.vue` into the default outlet and `child@sidebar.vue` into the `sidebar` outlet. This has actually been possible in Vue Router for a long time; this release wires it up to Nuxt's file-based routing.

::note
`definePageMeta` is read from the default route file only, and per-view rendering modes aren't supported (the parent page's mode applies to the default view).
::

:read-more{to="/docs/guide/directory-structure/app/pages#named-views"}

> ## 🚦 `enabled` Option for `useAsyncData`

Reviewer: ```suggestion
## 🚦 `enabled` Option for `useFetch` and `useAsyncData`
```

after https://github.com/nuxt/nuxt/pull/35627 is merged into the release

You can now gate data fetching with a reactive `enabled` option ([#33260](https://github.com/nuxt/nuxt/pull/33260)). While `enabled` is `false`, every execution is blocked (the initial fetch, `execute`/`refresh`, and watch triggers), and if you flip it from `true` to `false` mid-flight, the in-flight request is cancelled without clearing your existing `data`.

```vue [app/pages/search.vue]
<script setup lang="ts">
const query = ref('')

> const { data } = await useAsyncData(
>   'search',
>   () => $fetch('/api/search', { query: { q: query.value } }),
>   {
>     // Only fetch once the user has typed something
>     enabled: () => query.value.length > 2,
>   },
> )
> </script>

Reviewer: ```suggestion
<script setup lang="ts">
const query = ref('')

const { data } = await useFetch('/api/search', {
  query: { q: query },
  // Only fetch once the user has typed something
  enabled: () => query.value.length > 2,
})
</script>
```

```

This is perfect for dependent or conditional queries, where you don't want to fire a request until some precondition is met. It pairs naturally with a getter or a ref, so it stays reactive.

::note

> This option currently applies to `useAsyncData`. Support in `useFetch` is [tracked separately](https://github.com/nuxt/nuxt/pull/33260); for now, reach for `useAsyncData` with `$fetch` when you need to gate a request.

Author: ```suggestion
This option currently applies to `useAsyncData`. Support in `useFetch` is [tracked separately](https://github.com/nuxt/nuxt/issues/29788); for now, reach for `useAsyncData` with `$fetch` when you need to gate a request.
```
